### PR TITLE
Add coco2014-mini: a simplified version of COCO for unit tests

### PR DESCRIPTION
**Pull request by marcenacp**

### CODE REVIEW

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract_test.py [hunk 1]
```diff
@@ -5,5 +5,5 @@
 
 
 def test_str_representation():
```
Comment (ccl-core) [inline python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract_test.py:7]: I'm not familiar with the codebase, but does this mean that we are testing only with empty files? Wouldn't it make sense at some point to also test w. dummy non-empty file(set)s?

Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract_test.py:7]: In this PR, I add datasets/coco2014-mini/data/train2014.zip. So we do test on non empty files.

This test is for the string representation.

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py [hunk 2]
```diff
@@ -18,9 +19,28 @@
 import pandas as pd
 
 
+def should_extract(encoding_format: str) -> bool:
```
Comment (ccl-core) [inline python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py:22]: NIT: more than should_extract is seems is_supported_encoding or something similar?

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py [hunk 3]
```diff
@@ -44,9 +63,14 @@ def __call__(self):
                     included_files.append(epath.Path(basepath) / file)
         # We need to sort `files` to have a deterministic/reproducible order.
         included_files.sort()
+        fullpaths = [
```
Comment (ccl-core) [inline python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py:66]: NIT: Given that this is a long list comprehension, you might want to add a comment to increase readibility (e.g. why is "[1:]" needed?)

**Review by ccl-core — APPROVED** [review]
Great, thanks Pierre!

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py (hunk 2 above)
Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py:22]: Acknowledged.

This function is used to know whether to extract or not a node. So I updated the docstring.

File: python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py (hunk 3 above)
Comment (marcenacp) [reply python/ml_croissant/ml_croissant/_src/operation_graph/operations/extract.py:66]: Done.

**Pull request merged**